test/specs/productDetails.js

```diff
@@ -1,29 +1,199 @@
+            await ProductsPage.btnAddBackpack.click();
+
+            expect(await ProductsPage.btnAddBackpack.isDisplayed()).toBe(false);
+            expect(await ProductsPage.btnRemoveBackpack.isDisplayed()).toBe(true);
+            expect(await ProductsPage.shoppingCartLabel.getText()).toEqual("1");
+
+            await ProductsPage.btnRemoveBackpack.click();
+
+            expect(await ProductsPage.shoppingCartLabel.isDisplayed()).toBe(false);
+            });
+
+        //TODO: sa continui de aici cu adaugarea testelor
+    })
+
+    describe('Footer Tests', () => {
+
+        beforeEach(async function () {
+            await LoginPage.open();
+            await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
+            await ProductsPage.allItemsImages[0].click()
+        });
+
+        afterEach(async function () {
+            await ProductsPage.logout();
         });
+
+        it("check Footer UI elements", async () => {
+
+            expect(await ProductsPage.twitterIcon.isDisplayed()).toBe(true);
+            expect(await ProductsPage.facebookIcon.isDisplayed()).toBe(true);
+            expect(await ProductsPage.linkedInIcon.isDisplayed()).toBe(true);
+            expect(await ProductsPage.copyright.isDisplayed()).toBe(true);
+            expect(await ProductsPage.copyright.getText()).toContain("© 2022 Sauce Labs. All Rights Reserved. Terms of Service | Privacy Policy");
+            expect(await ProductsPage.robotFooter.isDisplayed()).toBe(true);
+            expect(await ProductsPage.robotFooter.getAttribute('src')).toEqual(ImageSource.footerRobot);
+        });
+
+        it("the 'Twitter' icon should redirect to the Sauce Labs Twitter url", async () => {
+            await ProductsPage.twitterIcon.waitForClickable()
+            await ProductsPage.twitterIcon.click()
+            await browser.switchWindow("https://twitter.com/saucelabs")
+
+            expect(await browser.getUrl()).toEqual("https://twitter.com/saucelabs");
+            await ProductsPage.closeWindowAndSwitchBack()
+        })
+
+        it("the 'Facebook' icon should redirect to the Sauce Labs Facebook url", async () => {
+            await ProductsPage.facebookIcon.waitForClickable()
+            await ProductsPage.facebookIcon.click()
+            await browser.switchWindow("https://www.facebook.com/saucelabs")
+
+            expect(await browser.getUrl()).toEqual("https://www.facebook.com/saucelabs");
+            await ProductsPage.closeWindowAndSwitchBack()
+        })
+
+        it("the 'LinkedIn' icon should redirect to the Sauce Labs LinkedIn url", async () => {
+            await ProductsPage.linkedInIcon.waitForClickable()
+            await ProductsPage.linkedInIcon.click()
+            await browser.switchWindow("https://www.linkedin.com/company/sauce-labs/")
+
+            expect(await browser.getUrl()).toEqual("https://www.linkedin.com/company/sauce-labs/");
+            await ProductsPage.closeWindowAndSwitchBack()
+        })
```

Comment:
same as the burger menu tests. you don't have to check the footer functionalities for every page, unless there are known differences between pages.

Reply:
the file footerIconsFunctionalityTests has been added into the specs folder and the modifications have been made in products and productDetails